CMakeLists.txt

```diff
@@ -6,9 +6,24 @@ project(everest-timer
     LANGUAGES CXX C
 )
 
-find_package(everest-cmake 0.1 REQUIRED
-    PATHS ../everest-cmake
-)
+
+if (USE_FIND_PACKAGE)
+    find_package(everest-cmake 0.1 REQUIRED
+        PATHS ../everest-cmake)
+else()
+    include(FetchContent)
+
+    set(EXTERNAL_EVEREST_CMAKE_NAME everest-cmake)
+    set(EXTERNAL_EVEREST_CMAKE_URL https://github.com/EVerest/everest-cmake.git)
+    set(EXTERNAL_EVEREST_CMAKE_TAG v0.1.0)
+
+    message(STATUS "Fetching everest cmake")
+    FetchContent_Declare(
+        ${EXTERNAL_EVEREST_CMAKE_NAME}
+        GIT_REPOSITORY              ${EXTERNAL_EVEREST_CMAKE_URL}
+        GIT_TAG                     ${EXTERNAL_EVEREST_CMAKE_TAG}
+    )
+endif()
```

Comment:
Most of our "everest" libraries depend on everest-cmake because of the evc-setup macro.  I see, that it might be convenient, to put this `if/else` in place and use fetch_content to retrieve everest-cmake if it isn't available locally.  But actually, this should be the responsibility of the *builder*, not of this library.
In the end, it leads to a lot of bloat in this `CMakeLists.txt`, and it doesn't scale because for each external dependency you need to have this additional `if/else` which does fetch_content.  And the *builder* still needs to be aware of the `EXTERNAL_*` variables.
I guess, you don't want to use EDM, so in this case use the `USE_FIND_PACKAGE` flag and manually install the required dependencies (everest-cmake and date), like it is done in the everest yocto layer.  With that, libtimer should build fine. But I wouldn't add this responsibility into this library, because if we do so, there will be a lot of duplication and it will get very hard to maintain.